tests/src/Unit/AuthenticatedClientTest.php

```diff
@@ -210,6 +211,32 @@ public function testGetConfig()
         $this->assertEquals('the-value', $authenticatedClient->getConfig('the-option'));
     }
 
+    /**
+     * Test setting a duration for renewed tokens.
+     *
+     * @covers ::setTokenDuration()
+     * @covers ::mergeAuth()
+     */
+    public function testTokenDuration()
+    {
+        /** @var Client|\PHPUnit\Framework\MockObject\MockObject $client */
+        $client = $this->getMockBuilder(Client::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        /** @var UserSession|\PHPUnit\Framework\MockObject\MockObject $userSession */
+        $userSession = $this->getMockBuilder(UserSession::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $authenticatedClient = new AuthenticatedClient($client, $userSession);
+        $duration = rand(1, 3600);
+        $userSession->expects($this->once())->method('acquireToken')
+            ->with($duration, false)
+            ->willReturn(new Token('mpx/USER-ID', 'abcdef', $duration));
+        $authenticatedClient->setTokenDuration($duration);
+        $authenticatedClient->request('GET', 'https://www.example.com/');
```

Comment:
I don't see how this test works, no assertions?

Reply:
Line 233 has the assertion through the `expects()` call. If the method is called a different number of times, or with different parameters, an assertion will fail. You can try this locally by reverting the change in the `acquireToken` method call.

Reply:
:+1: